Error 50x styling updates

indirectlylit opened the pull request:
* uses table CSS instead of flexbox
* loads Noto Sans from online API when available; otherwise falls back on `sans-serif`
* updates some language
* lots of visual changes as shown below

| before | after |
|--|--|
|  ![image](https://user-images.githubusercontent.com/2367265/67127058-635a2880-f1ad-11e9-975e-0d15007b1dc6.png)  | ![image](https://user-images.githubusercontent.com/2367265/67126743-b2538e00-f1ac-11e9-9f72-b419c02067f1.png) |

indirectlylit committed:
update title, remove commented-out code

indirectlylit committed:
update spinner

indirectlylit committed:
tighten up text and styling

indirectlylit committed:
reference online Noto fonts, and fall back on sans serif when not available

indirectlylit commented:
builds off #35

jredrejo reviewed (commented):
@indirectlylit related to the aesthetic changes I have not any objection, I don't have a strong artistic side so, as far as it's not ugly,  if you like it more, it's fine for me. However I feel curiosity to know the reason of some of the changes you did:

1. You changed the strings, they are not the same that appeared in the approved mocks
2. Why did you move the animation part in the svg from the html tags to a css? Is there any technical reason for it?
3. Using external links to load fonts will block the load of this page if Internet is not available. If you want to include external fonts I suggest to use JavaScript to load them. That will require the use of the https://github.com/typekit/webfontloader library, what will add more complexity to this. I am not sure if it's worth adding it to have these fonts. I'd rather go for a solution having them in a static folder if you consider they are completely necessary.
4. Why did you change from flexbox to table css? flexbox is supported in all the browsers Kolibri supports and the code is more ellegant and clear (IMO)

indirectlylit commented:
> You changed the strings, they are not the same that appeared in the approved mocks

The previous strings were a bit non-committal and verbose. I tried to tighten it up a bit:

1. Made the first sentence more assertive and confident: "You should be..." rather than "If you are not..."
2. Removed the guidance to "try refreshing the page" because the page will be automatically refreshing, so that doesn't actually help
3. Removed redundant references to the community forums

> Why did you move the animation part in the svg from the html tags to a css? Is there any technical reason for it?

I copied the output code from our existing [spinner component in Kolibri](https://github.com/learningequality/kolibri/blob/develop/packages/kolibri-components/src/loaders/KCircularLoader.vue). The styling should match more closely now (except for the color).

> Using external links to load fonts will block the load of this page if Internet is not available. If you want to include external fonts I suggest to use JavaScript to load them. That will require the use of the https://github.com/typekit/webfontloader library, what will add more complexity to this. I am not sure if it's worth adding it to have these fonts. I'd rather go for a solution having them in a static folder if you consider they are completely necessary.

Would it help to move them to the bottom of the page?

The fonts are really tricky to get right. In Kolibri we have [complex tooling](https://github.com/learningequality/kolibri/blob/develop/build_tools/i18n/fonts.py) for generating custom fonts based on the text that gets displayed in all languages.

If we did move the page generation logic to the Kolibri repo as @rtibbles [suggested](https://github.com/learningequality/kolibri-server/pull/35#discussion_r337204062), things would be a bit easier because we could generate custom fonts for the server pages too...

> Why did you change from flexbox to table css? flexbox is supported in all the browsers Kolibri supports and the code is more ellegant and clear (IMO)

I agree that flexbox is more elegant, but we've tried to avoid flexbox by convention in Kolibri to support older Android tablets (below 4.4 has poor flexbox support). For an error page like this, I think it's preferable to err on the side of more browser compatibility (e.g. the comment about `let` vs `var`).

One day we may relax the flexbox constraint but for now this is our convention. (see https://kolibri-dev.readthedocs.io/en/develop/frontend_architecture/conventions.html#styling-anti-patterns)

jredrejo commented:
> Would it help to move them to the bottom of the page?
> 
> The fonts are really tricky to get right. In Kolibri we have [complex tooling](https://github.com/learningequality/kolibri/blob/develop/build_tools/i18n/fonts.py) for generating custom fonts based on the text that gets displayed in all languages.
> 
> If we did move the page generation logic to the Kolibri repo as @rtibbles [suggested](https://github.com/learningequality/kolibri-server/pull/35#discussion_r337204062), things would be a bit easier because we could generate custom fonts for the server pages too...
> 

Honestly, I am against moving to the kolibri code this. This code is only for a specific case use. It's useless in many other cases (windows  users, apache users, etc.)
I don't see the problem in having these (few) strings in its own project. I suppose the Windows installer strings are not in the kolibri repo either. I will eat these words if I am wrong ;)

> > Why did you change from flexbox to table css? flexbox is supported in all the browsers Kolibri supports and the code is more ellegant and clear (IMO)
> 
> I agree that flexbox is more elegant, but we've tried to avoid flexbox by convention in Kolibri to support older Android tablets (below 4.4 has poor flexbox support). For an error page like this, I think it's preferable to err on the side of more browser compatibility (e.g. the comment about `let` vs `var`).
> 
> One day we may relax the flexbox constraint but for now this is our convention. (see https://kolibri-dev.readthedocs.io/en/develop/frontend_architecture/conventions.html#styling-anti-patterns)

Thanks, I didn't know this. Good to be aware.

jredrejo reviewed (changes requested):
> Would it help to move them to the bottom of the page?

I have checked it, using the network throttling capabilities of the browsers:

With Firefox:
- Without Internet, the page loads fast, it discards the font
- With throttling, the page loads fast too, it ignores the font and does not block the page rendering.


With Chrome:
- Without Internet, the page loads fast, it discards the font
- With throttling, the page does not render until the font is loaded, and it can take long times
- Moving the font link to the bottom of the page solves the problem with throttling: the page renders fast and applies the font when it is loaded. I know this is an effect to avoid with css but I think this is a special case and we should do it that way.

@indirectlylit  if you agree, I will merge this PR and move the link to the bottom.

indirectlylit commented:
The behaviors above after moving to the bottom sound reasonable to me. In the future we can revisit

carinediaz commented:
Hi guys, @indirectlylit @khangmach @jamalex @rtibbles @jredrejo,

I just saw this. Needless to say, I'm puzzled as to why we're using this bird, cute as it is but totally off-brand, as opposed to our logo? Could I please request that if we're going to use a bird image, we use the official logo, otherwise this will be confusing from a branding perspective. Thank you!

jredrejo commented:
> Hi guys, @indirectlylit @khangmach @jamalex @rtibbles @jredrejo,
> 
> I just saw this. Needless to say, I'm puzzled as to why we're using this bird, cute as it is but totally off-brand, as opposed to our logo? Could I please request that if we're going to use a bird image, we use the official logo, otherwise this will be confusing from a branding perspective. Thank you!

No idea. I am blind to these things, @khangmach ?

benjaoming commented:
@carinediaz this page is deliberately trying to NOT look like an attempted theme, because the static assets for proper "branded" theming aren't available for an error page. At the same time, it tries to be friendly, hence the little bird icon. I think it's a good idea to stay clear of branding, and also to avoid making a page that is heavy in aspects of theming, interaction and programmatics.

The error page should not cause errors in itself.

Finally, you could argue that you don't want the brand associated with errors :)

carinediaz commented:
HI Ben, @benjaoming
@indirectlylit was just talking to me about the same thing. You guys are right, I hadn't thought about it from the negative POV and I agree, our logo should stay clear of it. I take it all back. Apologies, everyone.

jamalex commented:
Just adding a couple thoughts/data points (not prescriptions) here.

>Finally, you could argue that you don't want the brand associated with errors :)

In most cases (ideally!), this page is serving as a _loading_ page, rather than an actual _error_ page (and @indirectlylit asked me to look to this page as a model for the Flatpak loading page, as well -- so it could be good for us to standardize). I think it's fairly common to use one's own logo on a loading page, whereas error pages (e.g. 404 pages) often do include funkier/more playful elements.

The image pixelation could be seen as playful, but to me it kind of just looks low-quality/sloppy, or as if we were trying to go for a Windows XP loading screen aesthetic:
![image](https://user-images.githubusercontent.com/618416/68230293-9bcf7400-ffad-11e9-97a0-e139da4d36e2.png)

Another consideration is that given how many different "Kolibri" hummingbird logos there are, there are confusion risks of using something that is much closer to their brands than it is to ours:
![image](https://user-images.githubusercontent.com/618416/68229535-4f376900-ffac-11e9-8181-fe00a120f25f.png)
Source: https://www.google.com/search?tbm=isch&q=kolibri+logo

indirectlylit commented:
> this page is deliberately trying to NOT look like an attempted theme, because the static assets for proper "branded" theming aren't available for an error page

This is not true. The whole point of this PR was to bring this page more in line with our style guide, even given the limitations of not having the Kolibri assets available.

> Finally, you could argue that you don't want the brand associated with errors 

yes

> In most cases (ideally!), this page is serving as a loading page, rather than an actual error page

also yes

> The image pixelation could be seen as playful, but to me it kind of just looks low-quality/sloppy,

Agreed - will add to https://github.com/learningequality/clearinghouse/issues/413

> Another consideration is that given how many different "Kolibri" hummingbird logos there are, there are confusion risks of using something that is much closer to their brands than it is to ours

it's an interesting point - hadn't considered that. My sense is that for true error pages (e.g. Kolibri's error page), we have more liberty to stray from the official logo. But since this is in some sense a loading page we're probably better off sticking to the basics.

We have a pure SVG, animated, inline logo here that we can use here: https://github.com/learningequality/kolibri/pull/4833

carinediaz commented:
Wow, you guys are fast. I need to read more about it, thanks for the references @jamalex and @indirectlylit. But from this last bits of convo: 
1) loading page: Kolibri logo 
2) error page -  a more fun and on brand strategy? Devon shared this with me https://optinmonster.com/best-404-page-examples/ please check Lego's

indirectlylit commented:
opened follow-up PR here to address feedback: https://github.com/learningequality/kolibri-server/pull/40

@jamalex you can use this for the Flatpack work also
